src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -148,6 +152,31 @@ public function onKernelController(ControllerEvent $event): void
 
         $request->attributes->set('_mounted_component', $mounted);
 
+        // autowire live file arguments
+        if ($request->files->count()) {
+            $allFiles = $request->files->all();
+            $accessor = PropertyAccess::createPropertyAccessor();
+            foreach (LiveFileArg::liveFileArgs($component, $action) as $parameter => $fileArg) {
+                $path = $fileArg->getPropertyPath();
+
+                if (!$accessor->isReadable($allFiles, $path)) {
+                    throw new \RuntimeException(sprintf('File path "%s" for parameter "%s" is invalid', $fileArg->name, $parameter));
```

Comment:
```suggestion
                    throw new \RuntimeException(sprintf('The LiveFileArg name "%s" for argument "%s" was not found in the uploaded data. Uploaded file keys were: (%s)', $fileArg->name, $parameter, implode(', ', array_keys($allFiles)));
```

Reply:
And, do we need this at all? If they have a `LiveFileArg` but it is (for some reason) optional, then I don't think we need to throw an error.

Reply:
I think we don't. I tried here to handle all cases that come from ProeprtyAccessor. But I think with your idea of the changes it will become obsolete. When we have `data-name` and `name` fallback I think we just can assume one of them will be simple (not `form[data][file]` type), so it will get much simpler.